src/SW_netCDF_Input.c

```diff
@@ -0,0 +1,3540 @@
+/* --------------------------------------------------- */
+
+/*
+@brief Translate an input keys into indices the program can understand
+
+@param[in] varKey Read-in variable key/category from input file
+@param[in] varName Read-in variable SW2 variable name from input file
+@param[out] inKey Translated key from read-in values to local arrays
+@param[out] inVarNum Translated key from read-in values to local arrays
+@param[out] isIndex Specifies that the read-in values were related to an index
+file
+@param[out] isAllVegVar Specifies that the read-in values were a generalization
+of veg or soil variables
+*/
+static void get_2d_input_key(
+    char *varKey,
+    char *varName,
+    int *inKey,
+    int *inVarNum,
+    Bool *isIndex,
+    Bool *isAllVegVar
+) {
+
+    int keyNum;
+    int varNum;
+    const int numGeneralVegNames = 5;
+
+    *inKey = eSW_NoInKey;
+    *inVarNum = KEY_NOT_FOUND;
+    *isIndex = swFALSE;
+    *isAllVegVar = swFALSE;
+
+    for (keyNum = 0; keyNum < SW_NINKEYSNC && *inKey == eSW_NoInKey; keyNum++) {
+        if (strcmp(varKey, possInKeys[keyNum]) == 0) {
+            *inKey = keyNum;
+        }
+    }
+
+    if (*inKey != eSW_NoInKey) {
+        for (varNum = 0; varNum < numVarsInKey[*inKey]; varNum++) {
+            if (strcmp(possVarNames[*inKey][varNum], varName) == 0) {
+                if (varNum == 0 && keyNum != eSW_InDomain &&
+                    keyNum != eSW_InSpatial) {
+
+                    *isIndex = swTRUE;
+                }
+
+                *inVarNum = varNum;
+                return;
+            }
+        }
+
+        if (*inKey == eSW_InVeg) {
+            for (varNum = 0; varNum < numGeneralVegNames; varNum++) {
+                if (strcmp(generalVegNames[varNum], varName) == 0) {
+                    *isAllVegVar = swTRUE;
+                    *inVarNum = varNum;
+                    return;
+                }
+            }
+        } else if (*inKey == eSW_InSoil) {
+            if (strcmp(generalSoilNames[0], varName) == 0) {
+                *isAllVegVar = swTRUE;
+                *inVarNum = 0;
+            }
+        }
+    }
+}
+
+/**
+@brief Check that the required variable information was input through
+the input variable file
+
+@param[in] inputInfo Attribute information for a specific input variable
+@param[in] inWeathStrideInfo List of stride information for weather variables
+@param[in] key Current category of input variables being tested
+@param[in] varNum Variable number within the given key
+@param[out] LogInfo Holds information on warnings and errors
+*/
+static void check_variable_for_required(
+    char ***inputInfo,
+    const int inWeathStrideInfo[],
+    int key,
+    int varNum,
+    LOG_INFO *LogInfo
+) {
+
+    int attNum;
+    int mustTestAttInd[] = {
+        SW_INNCVARNAME,
+        SW_INGRIDTYPE,
+        SW_INCRSNAME,
+        SW_INCRSEQUIV,
+        SW_INXAXIS,
+        SW_INYAXIS
+    };
+    const char *mustTestAttNames[] = {
```

Comment:
Please update all code to handle the recent changes to the input.tsv including 
- remove: ncGridType, ncCRSEquivalency
- new: ncDomainType, ncSiteName, ncCRSGridMappingName